File: CS/EditReferenceProperiesInBatchEditMode.Module/BusinessObjects/SimpleDemo.cs
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Xpo;

namespace EditReferenceProperiesInBatchEditMode.Module.BusinessObjects {
    [DefaultClassOptions]
    public class SimpleDemo : BaseObject {
        public SimpleDemo(Session session) : base(session) { }
        private string name;
        public string Name {
            get {
                return name;
            }

            set {
                SetPropertyValue<string>(nameof(Name), ref name, value);
            }
        }
        private SimpleData data;
        public SimpleData Data {
            get {
                return data;
            }

            set {
                SetPropertyValue(nameof(Data), ref data, value);
            }
        }
    }
    public class SimpleData : BaseObject {
        public SimpleData(Session session) : base(session) { }
        private string name;
        public string Name {
            get {
                return name;
            }

            set {
                SetPropertyValue<string>(nameof(Name), ref name, value);
            }
        }
    }
}

File: VB/EditReferenceProperiesInBatchEditMode.Module/BusinessObjects/SimpleDemo.vb
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Xpo

<DefaultClassOptions>
Public Class SimpleDemo
    Inherits BaseObject

    Public Sub New(ByVal session As Session)
        MyBase.New(session)
    End Sub

    Private _name As String

    Public Property Name As String
        Get
            Return _name
        End Get
        Set(ByVal value As String)
            SetPropertyValue(Of String)(NameOf(Name), _name, value)
        End Set
    End Property

    Private _data As SimpleData

    Public Property Data As SimpleData
        Get
            Return _data
        End Get
        Set(ByVal value As SimpleData)
            SetPropertyValue(NameOf(Data), _data, value)
        End Set
    End Property
End Class

Public Class SimpleData
    Inherits BaseObject

    Public Sub New(ByVal session As Session)
        MyBase.New(session)
    End Sub

    Private _name As String

    Public Property Name As String
        Get
            Return _name
        End Get
        Set(ByVal value As String)
            SetPropertyValue(Of String)(NameOf(Name), _name, value)
        End Set
    End Property
End Class
